File: src/main/java/com/graqr/threshr/model/redsky/products/Item.java
package com.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record Item(
        @Nullable @JsonProperty("cart_add_on_threshold")
        Double cartAddOnThreshold,

        Compliance compliance,

        @Nullable Disclaimer disclaimer,

        String dpci,

        @JsonProperty("eligibility_rules")
        EligibilityRules eligibilityRules,

        Enrichment enrichment,

        @Nullable @JsonProperty("environmental_segmentation")
        EnvironmentalSegmentation environmentalSegmentation,

        @Nullable @JsonProperty("formatted_return_method")
        String formattedReturnMethod,

        Fulfillment fulfillment,

        @Nullable Handling handling,

        @Nullable @JsonProperty("is_fresh_grocery")
        Boolean isFreshGrocery,

        @JsonProperty("is_limited_time_offer")
        Boolean isLimitedTimeOffer,

        @JsonProperty("merchandise_classification")
        MerchandiseClassification merchandiseClassification,

        @Nullable @JsonProperty("package_dimensions")
        PackageDimensions packageDimensions,

        @Nullable @JsonProperty("primary_barcode") String primaryBarcode,

        @JsonProperty("primary_brand")
        PrimaryBrand primaryBrand,

        @JsonProperty("product_classification")
        ProductClassification productClassification,

        @JsonProperty("product_description")
        ProductDescription productDescription,

        @JsonProperty("product_vendors")
        List<ProductVendor> productVendors,

        @JsonProperty("relationship_type_code")
        String relationshipTypeCode,

        @Nullable @JsonProperty("return_policies_guest_message")
        String returnPoliciesGuestMessage,

        @Nullable List<String> ribbons,

        @Nullable @JsonProperty("wellness_merchandise_attributes")
        List<?> wellnessMerchandiseAttributes) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/Item.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Item(
    @param:Nullable @param:JsonProperty("cart_add_on_threshold") val cartAddOnThreshold: Double,
    val compliance: Compliance,
    @param:Nullable val disclaimer: Disclaimer,
    val dpci: String,
    @param:JsonProperty("eligibility_rules") val eligibilityRules: EligibilityRules,
    val enrichment: Enrichment,
    @param:Nullable @param:JsonProperty("environmental_segmentation") val environmentalSegmentation: EnvironmentalSegmentation,
    @param:Nullable @param:JsonProperty("formatted_return_method") val formattedReturnMethod: String,
    val fulfillment: Fulfillment,
    @param:Nullable val handling: Handling,
    @param:Nullable @param:JsonProperty("is_fresh_grocery") val isFreshGrocery: Boolean,
    @param:JsonProperty("is_limited_time_offer") val isLimitedTimeOffer: Boolean,
    @param:JsonProperty("merchandise_classification") val merchandiseClassification: MerchandiseClassification,
    @param:Nullable @param:JsonProperty("package_dimensions") val packageDimensions: PackageDimensions,
    @param:Nullable @param:JsonProperty("primary_barcode") val primaryBarcode: String,
    @param:JsonProperty("primary_brand") val primaryBrand: PrimaryBrand,
    @param:JsonProperty("product_classification") val productClassification: PrimaryBrand,
    @param:JsonProperty("product_description") val productDescription: ProductDescription,
    @param:JsonProperty("product_vendors") val productVendors: List<ProductVendor>,
    @param:JsonProperty("relationship_type_code") val relationshipTypeCode: String,
    @param:Nullable @param:JsonProperty("return_policies_guest_message") val returnPoliciesGuestMessage: String,
    @param:Nullable val ribbons: List<String>,
    @param:Nullable @param:JsonProperty("wellness_merchandise_attributes") val wellnessMerchandiseAttributes: List<Any>
) {}
